app/StyledApp.tsx

```diff
@@ -70,9 +70,9 @@ const StyledApp = () => {
             ))}
             <Redirect to="/auth" />
           </Switch>
+          <WriteFilePermissionError />
+          <NoInternetConnection />
         </Router>
```

Comment:
```suggestion
        </Router>
       <WriteFilePermissionError />
       <NoInternetConnection />
```
Because it is working through the React Portal, better to keep it out of the Router scope.

Reply:
Could you tell me why?
As I know, the whole `<Router>` component should not be re-rendered on `history.push` — so what is the issue?

Previously we had a "Back" button there — and it won't work without the react-router context.
Right now I've removed this button, so we don't use react-router context anymore.
However, I'm sure it will be required in the future (in other notifications, e.g. #1142).
And I believe it is better to have all such modals in the same context (to make maintenance/development easier).

Reply:
Ok, I undestand your point.